lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -612,6 +613,8 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
         // required access
         os << " SWIFT_UNAVAILABLE";
         skipAvailability = true;
+        // If -init is unavailable, then +new should be, too:
+        makeNewUnavailable = selectorPieces.size() == 1 && selectorPieces.front().str() == "init";
```

Comment:
Teeeechnically this would catch `init:` as well. A better check would be `selector.getNumArgs()`.

Reply:
So `selectorPieces.size() == 1 && selector.getNumArgs() == 0 && selectorPieces.front().str() == "init";`?

Reply:
`selector.getNumArgs()` should be sufficient. There's no way to have a multi-piece selector with zero arguments, and there are no zero-piece selectors.

Reply:
Ah duh of course. Ok fixed, thanks!
I have only run the tests for `PrintAsObjC` locally, I assume you have to tell CI to run for this?

Reply:
Yep, I got it.